packages/react/src/components/ComboBox/ComboBox.js

```diff
@@ -113,6 +113,7 @@ const ComboBox = (props) => {
   );
   const [prevSelectedItem, setPrevSelectedItem] = useState(null);
   const [doneInitialSelectedItem, setDoneInitialSelectedItem] = useState(null);
+  const { current: handleInputChange } = useRef(onInputChange);
```

Comment:
```suggestion
  const savedOnInputChange = useRef(onInputChange);
```

Reply:
I think we'll need to not destructure this so that the effect doesn't keep changing like in the previous implementation when passing in the prop.